src/lib/storage/StorageApi.ts

```diff
@@ -0,0 +1,269 @@
+  ): Promise<{ data: { message: string } | null; error: Error | null }> {
+    try {
+      if (!isBrowser()) throw new Error('No browser detected.')
+
+      const formData = new FormData()
+      formData.append('cacheControl', '3600')
+      formData.append('', file, file.name)
+
+      const res = await fetch(`${this.url}/object/${path}`, {
+        method: 'POST',
+        body: formData,
+        headers: { ...this.headers },
+      })
+
+      if (res.ok) {
+        const data = await res.json()
+        return { data, error: null }
+      } else {
+        const error = await res.json()
+        return { data: null, error }
+      }
+    } catch (error) {
+      return { data: null, error }
+    }
+  }
+
+  /**
+   * Update a file
+   * @param path the relative file path
+   * @param file the File content
+   */
+  async updateFile(
+    path: string,
+    file: File
+  ): Promise<{ data: { Key: string } | null; error: Error | null }> {
+    try {
+      if (!isBrowser()) throw new Error('No browser detected.')
+
+      const formData = new FormData()
+      formData.append('cacheControl', '3600')
+      formData.append('', file, file.name)
+
+      const res = await fetch(`${this.url}/object/${path}`, {
+        method: 'PUT',
+        body: formData,
+        headers: { ...this.headers },
+      })
+
+      if (res.ok) {
+        const data = await res.json()
+        return { data, error: null }
+      } else {
+        const error = await res.json()
+        return { data: null, error }
+      }
+    } catch (error) {
+      return { data: null, error }
+    }
+  }
+
+  /**
+   * Rename a file
+   * @param bucketName
+   * @param fromPath original relative file path
+   * @param toPath the new relative file path
+   */
+  async renameFile(
+    bucketName: string,
+    fromPath: string,
+    toPath: string
+  ): Promise<{ data: { message: string } | null; error: Error | null }> {
+    try {
+      const data = await post(
+        `${this.url}/object/rename`,
+        { bucketName, sourceKey: fromPath, destinationKey: toPath },
+        { headers: this.headers }
+      )
+      return { data, error: null }
+    } catch (error) {
+      return { data: null, error }
+    }
+  }
+
+  /**
+   * Create signed url to download file
+   * @param path the relative file path to be downloaded
+   * @param expiresIn seconds until the signed URL expires
+   */
+  async createSignedUrl(
+    path: string,
+    expiresIn: number
+  ): Promise<{ data: { signedURL: string } | null; error: Error | null }> {
+    try {
+      const data = await post(
+        `${this.url}/object/sign/${path}`,
+        { expiresIn },
+        { headers: this.headers }
+      )
+      return { data, error: null }
+    } catch (error) {
+      return { data: null, error }
+    }
+  }
+
+  /**
+   * Download a file
+   * @param path the relative file path to be downloaded
+   */
+  async downloadFile(path: string): Promise<{ data: Blob | null; error: Error | null }> {
```

Comment:
Thoughts on making this `download()`?

Reply:
Perhaps not actually - we have `deleteFile` below, which we can't turn into `delete` because it's a reserved word. For consistency and accuracy, `downloadFile()` is probably better